src/ResourceManager/Resources/Commands.Resources/ManagementPartner/RemoveManagementPartner.cs

```diff
@@ -11,13 +26,34 @@ public class RemoveManagementPartner : AzureManagementPartnerCmdletsBase
         [ValidateNotNull]
         public string PartnerId { get; set; }
 
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
         public override void ExecuteCmdlet()
         {
             if (ShouldProcess(string.Format(PartnerResources.RemoveManagementParnterTarget, PartnerId),
                 string.Format(PartnerResources.RemoveManagementParnterAction, PartnerId)))
             {
-                AceProvisioningManagementPartnerApiClient.Partner.DeleteAsync(PartnerId).Wait();
-                WriteObject(string.Format(PartnerResources.RemovedManagementPartner, PartnerId));
+                Exception exception = null;
```

Comment:
Opps. I didn't mean to write the exception. I meant that, I'm assuming DeleteAsync throws when it cannot delete a resource. So, this would simply be:
```cs
AceProvisioningManagementPartnerApiClient.Partner.DeleteAsync(PartnerId).Wait();
if (PassThru.IsPresent)
{
    WriteObject(true);
}
```
Then, nothing is written when -PassThru is not used and you succeed. Otherwise, if you fail, the exception propagates to the PowerShell session.

Reply:
Got you